### PR TITLE
Refactor simplify tests

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

move functions to helpers and create Simple regression and classification model

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/helpers/test_models.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        (None, BasicGAN),
+        (None, ParityModuleRNN),
+        (None, ParityModuleMNIST),
+        (ClassifDataModule, ClassificationModel),
+        (RegressDataModule, RegressionModel),
+    ]
+)
+def test_models(tmpdir, data_class, model_class):
+    """Test simple models"""
+    dm = data_class() if data_class else data_class
```
Comment (s-rog) [inline tests/helpers/test_models.py:37]: What is the expected output for `datamodule=None`?

**Review by s-rog — APPROVED** [review]

**Comment by carmocca** [comment]
What's the point of the `base` -> `helpers` rename?

File: tests/helpers/test_models.py (hunk 1 above)
Comment (Borda) [reply tests/helpers/test_models.py:37]: It just uses the dataloaders defined inside the lightning module

**Comment by Borda** [comment]
> What's the point of the `base` -> `helpers` rename?

I feel like it was confusing name and also separate the items which will go further from deprecated EvalModule which shall be removed, so what left in base shall be dropped

**Review by SkafteNicki — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Looks great ! Awesome contrib.

**Review by justusschock — APPROVED** [review]
